### ISSUE TITLE
publish multiple events from list

### ISSUE THREAD

**Issue by AChehre** [zarusz/SlimMessageBus#99]
I have a list of events 
`
        var events = new List<IDomainEvent>()
        {
            new DomainCreated(),
            new DomainStateChanged()
        };
`

And when I call publish method 
`
mb.Publish(events.First());
`

I get this error:
`
One or more errors occurred. (Message of type IDomainEvent was not registered as a supported produce message. Please check your MessageBus configuration and include this type or one of its base types.)
`
This is because I registered DomainCreated but not  IDomainEvent.

So for this kind of problem, I'm using MessageBusProxy.Target.Publish method to pass the type of message.
Is there any way to publish message without passing the type? or add a method to IMessageBus class to pass the type?

**Comment by AChehre** [comment]
> Hello, it should work as you have it. The type should be taken from the passed actual message type.
> 
> It might be a bug, and if so I can review and fix.
> 
> What version are you using? Can you post the snippet of your bus registration? What plugins you use?
Version is 1.6.0

Publisher bus registration
`
  services.AddScoped<IMessagePublisher, KafkaMessagePublisher>();
        services.AddSlimMessageBus((mbb, svp) =>
                                   {
                                       mbb.Produce<DomainModelCreated>(x => x.DefaultTopic(kafkaSetting.DefaultTopic))
                                          .WithSerializer(new JsonMessageSerializer())
                                          .WithProviderKafka(new KafkaMessageBusSettings(kafkaSetting.Brokers));
                                   });
`
Consumer registration
`
        services.AddSlimMessageBus((mbb, svp) =>
                                   {
                                         mbb
                                           .Produce<DomainModelCreated>(x => x.DefaultTopic(kafkaSetting.DefaultTopic))
                                           .Consume<DomainModelCreated>(x => x.Topic(kafkaSetting.DefaultTopic)
                                                                                .WithConsumer<DomainModelHandler>()
                                                                                .KafkaGroup($"{kafkaSetting.DefaultTopic}-consumer-group") //  Kafka provider specific
                                                                            //.SubscriptionName("some-azure-sb-topic-subscription") // Azure ServiceBus provider specific
                                                                           )
                                           .WithSerializer(new JsonMessageSerializer())
                                           .WithProviderKafka(new KafkaMessageBusSettings(kafkaSetting.Brokers));
                                   });
        services.AddMessageBusConsumersFromAssembly(typeof(DomainModelHandler).Assembly);
        services.AddMessageBusConfiguratorsFromAssembly(typeof(DomainModelHandler).Assembly);
`

**Comment by zarusz** [comment]
@AChehre to double-check, are the publisher and consumer two separate applications/processes? 
You cannot have  `.AddSlimMessageBus()` twice on the same DI. Internally it creates and starts the bus Kafka bus.

Also, you are missing the `DomainStateChanged` produce registration, so that will be another issue you will get.

That said, I still will look into the potential bug. I believe the reflection cache I am using does not account for interfaces of messages, and so I will have to extend it to support it.

**Comment by zarusz** [comment]
It must be a bug. I will try to address soon.

**Comment by zarusz** [comment]
@AChehre  this should fix the problem: https://www.nuget.org/packages/SlimMessageBus.Host.Kafka/1.18.2-rc1
Please let me know when you have a chance to verify.

I still need to add a few tests and do some more thinking and once this is done I will release a final version (not RC).

**Comment by zarusz** [comment]
@AChehre let me know if it worked for you. I would be ready to release a final version.

**Comment by AChehre** [comment]
> @AChehre let me know if it worked for you. I would be ready to release a final version.

Sorry for the delay, yes it's working. tnx.
